ArjunaCore/arjuna/classes/com/arjuna/ats/internal/arjuna/recovery/PeriodicRecovery.java

```diff
@@ -787,10 +845,24 @@ private void doWorkInternal()
             ClassLoader cl = switchClassLoader(m);
             try {
             m.periodicWorkSecondPass();
+            if (m instanceof SuspendBlockingRecoveryModule) {
+                /*
+                 * Once Recovery Modules (implementing SuspendBlockingRecoveryModule) indicate
+                 * they do not want to block recovery (i.e. there is no more work to do), they
+                 *  will never change to wanting to block recovery in later recovery cycles.
+                 * In other words, once a Recovery Module
+                 * (implementing SuspendBlockingRecoveryModule) switches from
+                 * `hasWork() == true` to `hasWork() == false`, it cannot
+                 * change its mind.
```

Comment:
Remark: See my earlier comments about forcing modules to remember that they'd previously reported that there was no more work to do.